wire-moshi-support/src/main/java/com/squareup/wire/WireJsonAdapterFactory.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.Moshi;
+import com.squareup.moshi.Types;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Set;
+import javax.annotation.Nullable;
+import okio.ByteString;
+
+/**
+ * A {@link JsonAdapter.Factory} that allows Wire messages to be serialized and deserialized using
+ * the Moshi Json library.
+ *
+ * <pre>
+ * Moshi moshi = new Moshi.Builder()
+ *   .add(new WireJsonAdapterFactory())
+ *   .build();
+ * </pre>
+ *
+ * The resulting {@link Moshi} instance will be able to serialize and deserialize any Wire {@link
+ * Message} type, including extensions and unknown field values. The JSON encoding is intended to be
```

Comment:
I copied that from the `gson` module; I fixed it. For extensions, I'm not 100% sure but since I didn't see any code that would take care of that, I wrote we don't support them, and ignore unknown fields.

Reply:
We actually support extensions.